src/Compilers/Core/Portable/InternalUtilities/Hash.cs

```diff
@@ -339,12 +332,20 @@ internal static int GetFNVHashCode(string text)
         internal static int GetFNVHashCode(System.Text.StringBuilder text)
         {
             int hashCode = Hash.FnvOffsetBias;
+
+#if NETCOREAPP3_1_OR_GREATER
+            foreach (var chunk in text.GetChunks())
+            {
+                hashCode = CombineFNVHash(hashCode, chunk.Span);
+            }
+#else
             int end = text.Length;
 
             for (int i = 0; i < end; i++)
             {
                 hashCode = unchecked((hashCode ^ text[i]) * Hash.FnvPrime);
             }
+#endif
```

Comment:
The `#else` section looks basically identical to `CombineFNVHash`. What am I missing that is different here?

Reply:
In addition to Jared's clarifying question, it was not obvious to me how this change is an optimization. Could you clarify? Maybe a `// PERF` comment would be useful for the next person.
It would also be good to share what validation was done and what effect was observed.

Reply:
Note that **StringTable.cs** contains a note regarding the indexer performance, which is visible lower in this review. Not sure if one needs to be added here, considering `GetChunks()` doesn't exist on some frameworks.

Reply:
That all makes sense. Can we add a comment to that effect here so someone doesn't try to "fix" it later on?

Reply:
Now updated!